crates/consensus-logic/src/csm/ctl.rs

```diff
@@ -60,4 +60,15 @@ impl EventSubmitter for CsmController {
 
         Ok(())
     }
+
+    async fn submit_event_idx_async(&self, ev_idx: u64) -> anyhow::Result<()> {
+        let msg = CsmMessage::EventInput(ev_idx);
+        if self.csm_tx.send(msg).await.is_err() {
+            warn!(%ev_idx, "sync event receiver closed when submitting");
+        } else {
+            trace!(%ev_idx, "sent csm event input");
+        }
+
+        Ok(())
```

Comment:
This does not need to be a `Result` return type. There's no way this can return the `Error` variant.

```suggestion
    async fn submit_event_idx_async(&self, ev_idx: u64) {
        let msg = CsmMessage::EventInput(ev_idx);
        if self.csm_tx.send(msg).await.is_err() {
            warn!(%ev_idx, "sync event receiver closed when submitting");
        } else {
            trace!(%ev_idx, "sent csm event input");
        }
```